### PR TITLE
Add mobile styles for giving season banner

### PR DESCRIPTION
**Pull request by oetherington**
This PR adds mobile styles for the giving season banner:
Screenshot 2023-11-08 at 22 08 33
Screenshot 2023-11-08 at 22 08 47
Screenshot 2023-11-08 at 22 09 03
Screenshot 2023-11-08 at 22 09 19

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205911104831062) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
FYI @agnesstenlund

**Comment by agnesstenlund** [comment]
Looks great! 🙌  we should keep the sticky behaviour of the normal top bar though (not the whole banner, just the slim top bar, like what's live now)

https://github.com/ForumMagnum/ForumMagnum/assets/13235378/fd76e9f2-01c8-4e20-a8ff-bb42443a0808

**Comment by oetherington** [comment]
@agnesstenlund Thanks!

I think that's actually pretty difficult to do (we don't do it for desktop currently). The main problem is we'd need to work out some nice way to transition from the the whole big banner to just the top bar if you scroll slowly and don't trigger hiding the header.

Maybe that should be a follow-up task if you still think it's worth it?

**Comment by agnesstenlund** [comment]
> I think that's actually pretty difficult to do (we don't do it for desktop currently). The main problem is we'd need to work out some nice way to transition from the the whole big banner to just the top bar if you scroll slowly and don't trigger hiding the header.

yeah I was thinking they'd be completely disconnected and there wouldn't really be a nice transition–once you scroll all the way back up they'd "fit" again (the top bar would be the same red as the "unrelated" banner). Sarah made a point about how it's important being able to reach the top bar easily on mobile since the scroll back up there takes longer than on desktop (which was why she didn't want to implement this big top bar on mobile without keeping the normal top bar sticky)

**Comment by agnesstenlund** [comment]
I also added some small tweaks which made the bottom padding weird, but I didn't dare touch the height, so maybe you can help with that 😅 😬

**Comment by oetherington** [comment]
Ah, ok - I'll take a look but probably the whole thing needs to be rewritten so it could take a while

**Review by Will-Howard — APPROVED** [review]
ASTLAC! The non-fixed version really works surprisingly well, I thought transitioning between the large and small versions would look a lot more janky

File: packages/lesswrong/components/ea-forum/giving-portal/GivingSeasonHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    },
+    "& .ais-SearchBox-input": {
+      color: theme.palette.givingPortal.homepageHeader.light4,
+    },
+    "& .ais-SearchBox-input::placeholder": {
+      color: theme.palette.givingPortal.homepageHeader.light4,
+    },
+    "& .KarmaChangeNotifier-starIcon": {
+      color: theme.palette.givingPortal.homepageHeader.light4,
+    },
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/ea-forum/giving-portal/GivingSeasonHeader.tsx:122]: I can't take credit - this was @s-cheng

File: packages/lesswrong/components/ea-forum/giving-portal/GivingSeasonBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+const styles = (theme: ThemeType) => ({
+  root: {
+    ...givingSeasonImageBackground(theme, "bottom"),
+    position: "relative",
+    display: "flex",
+    alignItems: "center",
+    height: BANNER_HEIGHT,
+    padding: 20,
+    paddingTop: 0,
+    overflowX: "clip",
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/ea-forum/giving-portal/GivingSeasonBanner.tsx:28]: I think it was unintentional - I just pushed a commit that should make it a bit nicer

**Comment by agnesstenlund** [comment]
I like this a lot, agree it looks smoother than expected 👏  would give it a tiny bit more breathing room (more top-padding above title and slightly bigger gap in-between sections) otherwise perfect

**Comment by jpaddison3** [comment]
At a width of 412px, the cutoff of the week looks kinda broken to me. :two-cents: Agnes has approved though so ¯\\\_(ツ)_/¯

**Comment by agnesstenlund** [comment]
looks mostly fine to me! would just remove some bottom padding here when the width is really slim and we hide the weeks
Screenshot 2023-11-13 at 10 52 44

**Comment by oetherington** [comment]
Thanks! I've just pushed a commit to add a bit more spacing - I'm going to go ahead and merge now and we can make a new PR if we want more adjustments. I can't repro the issue at 412px.

**Pull request merged**